### PR TITLE
Todos: inference options to AIG, use AIG in EIG

### PR DESCRIPTION
**Pull request by jayelm**
I was writing some OED stuff which ended up modifying AIG/EIG according to Long's TODOs, so I decided to make a PR here.

One concern might be having to recreate an args object in EIG when handing off control to AIG, since the args dict for AIG needs a specific x and y. Not sure how expensive that is.

In terms of testing, I just diffed the output of the coin example across the changes I made, and the output was identical, but I haven't done anything else.

### CODE REVIEW

File: src/oed.wppl [hunk 1]
```diff
@@ -91,15 +95,14 @@ var EIG = function(args) {
...
-            var mPosterior = inferM2(function() {
-                var m2Name = mNameSample(), m2 = mFuncs[m2Name];
-                var ll = m2(x,y);
-                factor(ll);
-                return m2Name
-            });
-            var kl = KL(mPosterior, mPrior);
-            return kl;
+
+            return AIG({
```
Comment (longouyang) [inline src/oed.wppl]: This is a little inefficient -- we've already constructed the prior over models inside `EIG` but by calling `AIG` as a subroutine, we're going to recompute the prior for every `x` and `y`.

Can you roll back this part and we'll just merge in the AIG infer change?

Comment (jayelm) [inline src/oed.wppl]: Sure. Is this a decision i.e. "We're not going to call AIG as a subroutine in EIG" or a directive i.e. "Make AIG accept an optional model prior argument as an optimization"?

Comment (longouyang) [inline src/oed.wppl]: Directive -- we should make AIG accept an optional model prior argument as an optimization, but this is going to look a little different based on some cleanup work I'm doing, so let's table it for now.

**Comment by jayelm** [comment]
Done

**Comment by mhtess** [comment]
Nice!

On Jun 24, 2016, at 12:49, Long Ouyang <notifications@github.com<mailto:notifications@github.com>> wrote:

Merged #1https://github.com/mhtess/webppl-oed/pull/1.

##

**Pull request merged**